### PR TITLE
[REF] pos*, barcodes, web: stop using assets_backend in the pos 

### PR DESCRIPTION
**Pull request by sdegueldre**
pos*: l10n_ae_pos, l10n_co_pos, l10n_fr_pos_cert, l10n_gcc_pos,
l10n_in_pos, l10n_sa_pos, point_of_sale, pos_adyen, pos_discount,
pos_epson_printer, pos_epson_printer_restaurant, pos_hr,
pos_hr_restaurant, pos_loyalty, pos_mercury, pos_restaurant,
pos_restaurant_adyen, pos_restaurant_stripe, pos_sale, pos_sale_loyalty,
pos_sale_product_configurator, pos_six, pos_stripe

Previously, the pos assets included almost the entirety of the
assets_backend. Most of the contents of the assets_backend is completely
useless in the PoS, meaning that the PoS will load slower because it
loads much more JS than it needs. It also means that it gets all the
side effects of this bundle (global event listeners, among other things)
that we don't want.

This commit removes the dependency of the pos assets on the
assets_backend to solve these issues. The pos assets are now their own
bundle with only what is needed in the PoS.

As for the unit testing bundle, the same logic applies but
unfortunately, because the unit testing code from web that we want to
use (eg automatic cleanups, cleaning of registries, etc) depends on the
legacy code, we need to include a lot more files than would otherwise be
needed. This situation will probably be improved as legacy code is
removed from web, but in the mean time, it is not very important for
this bundle to be lean, as it's a test bundle and loading speed is less
important.

Linked to: https://github.com/odoo/enterprise/pull/40502
Linked to: odoo/upgrade#4626

### CODE REVIEW

File: addons/point_of_sale/static/src/app/report_service.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+export const reportService = {
+    dependencies: ["rpc", "user", "ui"],
+    start(env, { rpc, user, ui }) {
+        const reportActionsCache = {};
+        return {
+            async download(reportXmlId, active_ids) {
+                ui.block();
+                try {
+                    reportActionsCache[reportXmlId] ||= rpc("/web/action/load", {
```
Comment (kebeclibre) [inline addons/point_of_sale/static/src/app/report_service.js:14]: if you are sure that your action only depends on its bare name, and not, for instance, on the context

Comment (sdegueldre) [reply addons/point_of_sale/static/src/app/report_service.js:14]: There are only 2 report actions used in the pos currently and neither depend on the context, so I think it's fine for now. If this changes in the future we can always add more stuff to the cache key like the action service but this service is not supposed to be general purpose

File: addons/point_of_sale/static/src/js/Screens/PartnerListScreen/PartnerDetailsEdit.js [hunk 2]
```diff
@@ -19,13 +19,8 @@ export class PartnerDetailsEdit extends Component {
...
         };
-
-        onMounted(() => {
-            this.env.bus.on("save-partner", this, this.saveChanges);
-        });
-
-        onWillUnmount(() => {
-            this.env.bus.off("save-partner", this);
+        Object.assign(this.props.imperativeHandle, {
+            save: () => this.saveChanges(),
```
Comment (caburj) [inline addons/point_of_sale/static/src/js/Screens/PartnerListScreen/PartnerDetailsEdit.js:23]: This refactor on partner screen is out of place. It probably deserves a separate commit.

Comment (sdegueldre) [reply addons/point_of_sale/static/src/js/Screens/PartnerListScreen/PartnerDetailsEdit.js:23]: The reason it's in this commit is that because we stop using the legacy environment, it no longer contains a bus, causing the previous code to crash. I don't think this diff is substantial enough to merit its own commit but if you think so I'll extract it out.

Comment (caburj) [reply addons/point_of_sale/static/src/js/Screens/PartnerListScreen/PartnerDetailsEdit.js:23]: Oh I see. It's fine to keep it here. No need to extract.

**Comment by caburj** [comment]
@robodoo rebase-ff r+

**Comment by caburj** [comment]
@robodoo rebase-ff r+
